docs/build/guides/archival/choosing-the-right-storage.mdx

```diff
@@ -0,0 +1,151 @@
+---
+title: How to choose the right storage type for your use case
+hide_table_of_contents: true
+description: This guide walks you through choosing the most suitable storage type for your use case and how to implement it.
+---
+
+State archival refers to how data on the network is stored. This data includes all information that comprises the network's current and historical state. Storage is managed by keeping the important data active and archiving the rest in order to help keep the network fast and cheap. Read more about state archival [here](../../../learn/encyclopedia/storage/state-archival.mdx).
+
+State archival is vital for several reasons including:
+
+- It ensures that data is preserved efficiently and can be accessed when needed or restored in case of unexpected events;
+- It optimizes cost for users based on the data's importance and longevity;
+- It helps maintain optimal performance and throughput on the network.
+
+## Storage types
+
+When a smart contract is deployed on the Stellar network, the uploaded Wasm bytecode is stored as a `CONTRACT_DATA` entry on the ledger. Each entry has a stipulated time it can be accessed on the ledger before it is either archived or permanently deleted from the network; this is known as the entry's time to live (TTL). The TTL is measured in ledgers. That is, if an entry's TTL is 50, the entry will be available for the next 50 ledgers. When the entry's TTL reaches zero, it is either deleted or archived based on the type of storage used.
+
+There are three types of storage available on the Stellar network:
+
+- Temporary storage
+- Persistent storage
+- Instance storage
+
+### Temporary storage
+
+As the name implies, temporary storage stores data for a short period. It is suitable for contracts that only require their data for a short while and can afford for the data to be discarded afterward without any consequences. Here, once the TTL reaches zero, the data will be deleted from the ledger and cannot be restored. Temporary storage has unlimited storage space and the cheapest fees among all the storage types.
+
+Temporary storage is best suited for contracts with easily replaceable data and contracts that are only relevant within a certain period. For example, a time-bound auction contract where users can place bids, which only need to be kept for the duration of the auction.
```

Comment:
Another example or two of the type of data appropriate for temp storage would be really nice here! Not code, but just ideas to broaden the understanding of when it's appropriate.